guides/hack/21-XHP/04-setup.md

```diff
@@ -0,0 +1,65 @@
+## Namespace Support
+
+Support for namespaced XHP classes (elements) has only recently been added to
+HHVM (as of HHVM 4.68). Depending on the exact HHVM version, it may not be
```

Comment:
IMO we should strongly discourage use of v4 with whatever version of HHVM have the current defaults, and strongly discourage v3 after the default is flipped

Reply:
Heh I intentionally made it vague ("recently as of HHVM 4.68" -- though I guess ~4 months might be stretching the definition of "recently" a bit) because I wasn't sure what the exact version was, and it's probably impossible to pinpoint an exact version anyway (IIRC there have been some bugfixes and improvements after the support was first added...) but I agree, it would be better to clarify this.

That said, I also agree that we should discourage people from using XHP namespaces with HHVM versions that technically support them, but are not thoroughly tested (e.g. not used in production by any project), so I'd highly discourage using any HHVM version from before we converted at least the user-documentation site, and probably (like Fred) said, from before these flags are on by default.

So probably something like: "...has been added around HHVM 4.46, but hasn't been enabled by default until HHVM 4.xx, so we recommend using HHVM 4.xx [same xx] or newer. In older HHVM versions, modern XHP features including namespace support can be enabled by adding the following flags to your `.hhconfig`:" (also clarifies that some flags are technically unrelated to namespaces -- but given that we support/test either with none of them (xhp-lib v3) or with all of them (xhp-lib v4), I'd discourage people from using other subsets -- I also wrote that in the guide later)

Reply:
I'm also going to chagne xhp-lib's minimum HHVM version once the default flips, before 4.0.0 final - this is so that when people do `composer require facebook/xhp-lib`, they get a version that works with the default settings. This isn't currently an issue as composer will not install an rc version unless you explicitly request it.